managedplugin/metrics.go

```diff
@@ -1,10 +1,57 @@
 package managedplugin
 
-import "sync/atomic"
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"sync/atomic"
+)
+
+type DownloadMode int
```

Comment:
FWIW my little robot sidekick suggested `DownloadSource`. Not sold either way.

Reply:
`DownloadResult` ?

Reply:
`AssetSource`?

Reply:
I don't like `Result` because it sounds like something that would store a success or an error. `AssetSource` makes sense. I'll change it after I get some testing done.

Reply:
I changed this to `AssetSource`.